### DISCUSSION TITLE
selenium webdriver vs webdriver io

### DISCUSSION THREAD

**Discussion by spleshakov** [webdriverio/webdriverio#6963] [Q&A] [answered]
I'm trying to dig into differences between selenium webdriver and webdriver io, to understand which features each support if you compare them side by side (if you know of any GOOD articles please lmk, because existing ones are not great)

But the question is different - I was under impression that WDIO is an extension (wrapper) to `selenium webdriver` + JavaScript Language Bindings, meaning that it's used under the hood. A few unofficial webpages stated the same ([here](https://blog.kevinlamping.com/4-reasons-why-you-should-think-about-using-webdriverio#ithasthepowerofselenium) and [here](https://medium.com/@specktackle/selenium-and-webdriverio-a-historical-overview-6f8fbf94b418)). However, the `package.json` and `package-lock.json` files have no occurrences of either words.

is WDIO related to Selenium WebDriver? One is built on top of another?

**Comment by christian-bromann** [comment, marked as answer]
> is WDIO related to Selenium WebDriver?

It is not. These are two different projects that allow automation based on the [WebDriver](https://w3c.github.io/webdriver/) protocol.

> One is built on top of another?

WebdriverIO has implemented its own [protocol binding](https://www.npmjs.com/package/webdriver). Both projects don't share any code.

> to understand which features each support if you compare them side by side

I can't write down a whole article about the differences given my bias towards this projects. I believe something everyone would agree is that Selenium is an automation tool that you use to build your own automation framework out of (we have seen popular examples in the past where this happened, e.g. Protractor). The WebdriverIO project is not only an automation tool it is a full fledge test framework that allows you to have an opinionated setup with various of plugins for reporting and integration into other environments such as connecting to Sauce Labs etc.

If you have specific questions I am happy to answer these.
